### DISCUSSION TITLE
Contributing to Cluster Metrics

### DISCUSSION THREAD

**Discussion by N3uralN3twork** [online-ml/river#702] [Q&A] [answered]
I was wondering if there was a standard practice for contributing to the clustering metrics portion of this module. I've been doing research at my university for my thesis and it so happens to be on internal clustering metrics.

I've written a codebase full of metrics (about 30) that can be computed in much the same way as the current river library does.

Essentially, how do I help to contribute my current work to this project, as I would love to help out any way I can?

**Comment by MaxHalford** [comment, marked as answer]
Hello there, thanks for your interest.

We're never closed to contributions, of course. But it's important that we don't bloat the codebase and make sure it's accessible and, in a way, as minimal and useful as possible.

I'm not very comfortable with the amount of clustering metrics we currently have. There's already a lot. I also don't think people are using them that much. So adding 30 more sounds like too much.

Ideally, we would have a contrib package where we can dump a lot stuff without worrying about bloat. Then we can promote contrib code to the main codebase if we feel that it's popular/useful.

I'll try to create a contrib package released soon.

**Reply by MaxHalford** [reply]
The [`river_extra`](https://github.com/online-ml/river-extra) is now usable and installable.
